src/fides/api/ops/task/graph_task.py

```diff
@@ -746,16 +751,20 @@ def collect_tasks_fn(
             if not tn.is_root_node():
                 data[tn.address] = GraphTask(tn, resources)
 
+        # We don't rely on the output of `traverse` for the `end_nodes` like we do for
+        # the `access` task since all nodes are considered end nodes unless
+        # they are re-prioritized by the `erase_after` collection field
         env: Dict[CollectionAddress, Any] = {}
-        end_nodes = traversal.traverse(env, collect_tasks_fn)
+        traversal.traverse(env, collect_tasks_fn)
+        end_nodes = list(graph.nodes.keys())
 
-        def termination_fn(*dependent_values: int) -> Tuple[int, ...]:
-            """The dependent_values here is an int output from each task feeding in, where
-            each task reports the output of 'task.rtf(access_request_data)', which is the number of
-            records updated.
-
-            The termination function just returns this tuple of ints."""
-            return dependent_values
+        def termination_fn(*_: int) -> Dict[str, int]:
+            """
+            The erasure order can be affected in a way that not every node is directly linked
+            to the termination node. This means that we can't just aggregate the inputs directly,
+            we must read the erasure results from the cache.
+            """
+            return resources.get_all_cached_erasures()
```

Comment:
Major change: returning the erasure counts from the cache instead of the direct output of each erasure task